Support PHP 8

erik-zicht opened the pull request:
- [x] `use Doctrine\Persistence\Event\LoadClassMetadataEventArgs` in `StrictPublicUrlListener`
- [x] Rewrite `Params` for using `SortedSetMap`

7ochem commented on .gitignore:
```diff
@@ -1,5 +1,8 @@
 vendor/
 build/
-docker-compose.yml
+.env
```
.env heb je verwijderd toch?

erik-zicht replied:
Ehh ja. Dat was toch conform de afspraken?

7ochem replied:
Maar dan hoeft ie niet in de .gitignore?

7ochem replied:
Oooooohhh... Je hebt em wel staan lokaal, maar je wilt em niet in je repo 😅 Dat is ook het idee van .gitignore 🤦🏻‍♂️ 

Ik ga nog even wat koffie zetten denk ik 😆 ☕

erik-zicht replied:
:laughing:

erik-zicht commented on composer.json:
```diff
@@ -20,6 +20,9 @@
         "zicht/util": "^1.7"
     },
+    "conflict": {
+        "psr/container": ">1.1.1"
```
In `1.1.2` is deze commit geintroduceerd https://github.com/php-fig/container/commit/6009776263dfc7aaa8559fb1b3ab30be907ad1a8#diff-b76faa5405b0efea15f9586f4d772bc5e76cf5bc0f066fa821127eca94c8f583 en PHP 7.2 vindt het niet OK als https://github.com/symfony/dependency-injection/blob/4.4/Exception/ExceptionInterface.php opnieuw `\Throwable` implementeert.

Omdat we (tijdelijk) `--ignore-platform-reqs` gebruiken bij de `composer install` komt versie `1.1.2` mee.

7ochem replied:
Waarom gebruik je `--ignore-platform-reqs`?

7ochem replied:
Aha, prima 👍🏻 Leef je uit! 😄

7ochem commented on CHANGELOG.md:
```diff
@@ -8,6 +8,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ### Added|Changed|Deprecated|Removed|Fixed|Security
 Nothing so far
 
+## 5.1.4 - 2021-11-15
```
Ik zou em 5.2.0 dopen. PHP 8 support is een nieuwe feature

7ochem replied:
Was nu de consensus dat dit een nieuwe major versie ging worden? Dan zou het hier nog even moeten worden aangepast

erik-zicht replied:
Nee, hier is nog geen consensus over. Geldt voor de PR's in alle andere bundles ook.

erik-zicht committed:
Add support PHP 8

require dbal 2.13 for using its deprecated methods
require persistence for using LoadClassMetadataEventArgs instead of deprecated Common
require zicht/util 2
rewrite SortedSetMap in Params

erik-zicht commented on src/Url/Params/Params.php:
```diff
@@ -5,7 +5,7 @@
 namespace Zicht\Bundle\UrlBundle\Url\Params;
 
-use Zicht\Bundle\FrameworkExtraBundle\Util\SortedSetMap;
+use Doctrine\Common\Collections\ArrayCollection;
```
Deze hadden we er al uitgegooid, maar bleek hier nog te worden gebruikt :see_no_evil: Ik heb de methodes uit `SortedSetMap` overgenomen en van binnen aangepast.
